Dialog docs to site (#2306)
jpmorganchase/salt-ds: site/docs/components/dialog/accessibility.mdx

---
title:
  $ref: ./#/title
layout: DetailComponent
sidebar:
  exclude: true
data:
  $ref: ./#/data
---

### Best practices

#### Focus

The Dialog component will trap focus within the Dialog when open. This means that the user will not be able to tab out of the Dialog and focus will be set to the first focusable element within the Dialog. When the Dialog is closed, focus will be returned to the element that was focused before the Dialog was opened.

#### Initial focus

> You can control which element receives focus when the Dialog is opened by passing an `initialFocus` prop to the Dialog component. This prop accepts a number, which indicates the order in which each tabbable element within the Dialog should receive focus. If no element is found, focus will be set to the first focusable element within the Dialog. The `initialFocus` can also be a ref to an element within the Dialog.

Reviewer: ```suggestion
You can control which element receives focus when the Dialog is opened by passing an `initialFocus` prop to the Dialog component. This prop accepts a number, which indicates the index of the tabbable element within the Dialog that should receive focus. If no element is found, focus will be set to the first focusable element within the Dialog. The `initialFocus` can also be a ref to an element within the Dialog.
```
